spec/upload_to_s3_spec.rb

```diff
@@ -0,0 +1,168 @@
+require 'tempfile'
+require_relative './spec_helper'
+
+describe Fastlane::Actions::UploadToS3Action do
+  let(:client) { instance_double(Aws::S3::Client) }
+  let(:test_bucket) { 'a8c-wpmrt-unit-tests-bucket' }
+
+  before do
+    allow(Aws::S3::Client).to receive(:new).and_return(client)
+  end
+
+  # Stub head_object to return a specific content_length
+  def stub_s3_head_request(key, content_length)
+    allow(client).to receive(:head_object)
+      .with(bucket: test_bucket, key: key)
+      .and_return(Aws::S3::Types::HeadObjectOutput.new(content_length: content_length))
+  end
+
+  def stub_s3_file_exists(key)
+    stub_s3_head_request(key, 1)
+  end
+
+  def stub_s3_file_does_not_exist(key)
+    stub_s3_head_request(key, 0)
+  end
```

Comment:
You might as well refactor this as a single helper like this:
```suggestion
  def stub_s3_file_exists(key, exists)
    content_length = exists ? 1 : 0
    allow(client).to receive(:head_object)
      .with(bucket: test_bucket, key: key)
      .and_return(Aws::S3::Types::HeadObjectOutput.new(content_length: content_length))
  end
```